src/AWS.Deploy.Orchestration/SystemCapabilityEvaluator.cs

```diff
@@ -27,105 +28,161 @@ public class SystemCapabilityEvaluator : ISystemCapabilityEvaluator
         private readonly ICommandLineWrapper _commandLineWrapper;
         private static readonly Version MinimumNodeJSVersion = new Version(14,17,0);
 
-        public SystemCapabilityEvaluator(ICommandLineWrapper commandLineWrapper)
-        {
-            _commandLineWrapper = commandLineWrapper;
-        }
+        /// <summary>
+        /// How long to wait for the commands we run to determine if Node/Docker/etc. are installed to finish
+        /// </summary>
+        private const int CAPABILITY_EVALUATION_TIMEOUT_MS = 60000; // one minute
 
-        public async Task<SystemCapabilities> Evaluate()
-        {
-            var dockerTask = HasDockerInstalledAndRunning();
-            var nodeTask = GetNodeJsVersion();
+        /// <summary>
+        /// How long to cache the results of a VALID Node/Docker/etc. check 
+        /// </summary>
+        private static readonly TimeSpan DEPENDENCY_CACHE_INTERVAL = TimeSpan.FromHours(1);
```

Comment:
Open to feedback on the lengths of time.

Reply:
I'm not a fan of us having a fixed time. I prefer if we clear the cache once a user starts a deployment, or after a fixed time if they dont. If a customer messes anything up with their node or docker installations after a successful check if performed, they'd have to wait an hour for us to recheck or they'd have to kill the deploy tool instance which i think would cause confusion.

Reply:
From discussion, look at if we should tie the SystemCapabilityEvaluator lifetime to the session in server mode.

Reply:
In https://github.com/aws/aws-dotnet-deploy/pull/837/commits/b22ed89708bf1b4e05ec4b89d774f841be431331 I now clear the cache when the deployment starts from either CLI or server mode.